Frameworks/CoreFoundation/NumberDate.subproj/CFTimeZone.c

```diff
@@ -1447,63 +1191,54 @@ CFDataRef CFTimeZoneGetData(CFTimeZoneRef tz) {
     return tz->_data;
 }
 
-/* This function converts CFAbsoluteTime to (Win32) SYSTEMTIME
- * (Aleksey Dukhnyakov)
- */
-#if DEPLOYMENT_TARGET_WINDOWS
-BOOL __CFTimeZoneGetWin32SystemTime(SYSTEMTIME * sys_time, CFAbsoluteTime time)
-{
-    LONGLONG l;
-    FILETIME * ftime=(FILETIME*)&l;
-
-    /*  seconds between 1601 and 1970 : 11644473600,
-     *  seconds between 1970 and 2001 : 978307200,
-     *  FILETIME - number of 100-nanosecond intervals since January 1, 1601
-     */
-    l=(LONGLONG)(time+11644473600LL+978307200)*10000000;
-    if (FileTimeToSystemTime(ftime,sys_time))
-        return TRUE;
-    else
-        return FALSE;
-}
-#endif
-
 CFTimeInterval CFTimeZoneGetSecondsFromGMT(CFTimeZoneRef tz, CFAbsoluteTime at) {
     CFIndex idx;
     __CFGenericValidateType(tz, CFTimeZoneGetTypeID());
-    idx = __CFBSearchTZPeriods(tz, at);
-    return __CFTZPeriodGMTOffset(&(tz->_periods[idx]));
+
+    int32_t rawOffset;
+    int32_t dstOffset;
+    UErrorCode status = U_ZERO_ERROR;
+    tz->_timeZone->getOffset(__CFAbsoluteTimeToUDate(at), false, rawOffset, dstOffset, status);
+    if (U_SUCCESS(status)) {
+        // rawOffset is in milliseconds, CFTimeInterval is in seconds
+        return (rawOffset + dstOffset) / 1000;
+    }
+    return 0.0;
 }
 
 CFStringRef CFTimeZoneCopyAbbreviation(CFTimeZoneRef tz, CFAbsoluteTime at) {
-    CFStringRef result;
-    CFIndex idx;
     __CFGenericValidateType(tz, CFTimeZoneGetTypeID());
-    idx = __CFBSearchTZPeriods(tz, at);
-    result = __CFTZPeriodAbbreviation(&(tz->_periods[idx]));
-    return result ? (CFStringRef)CFRetain(result) : NULL;
+    UnicodeString result;
+    tz->_timeZone->getID(result);
+    return __UnicodeStringToCFStringRef(result);
 }
 
 Boolean CFTimeZoneIsDaylightSavingTime(CFTimeZoneRef tz, CFAbsoluteTime at) {
-    CFIndex idx;
     __CFGenericValidateType(tz, CFTimeZoneGetTypeID());
-    idx = __CFBSearchTZPeriods(tz, at);
-    return __CFTZPeriodIsDST(&(tz->_periods[idx]));
+
+    UErrorCode status = U_ZERO_ERROR;
+    UBool daylight = tz->_timeZone->inDaylightTime(__CFAbsoluteTimeToUDate(at), status);
+
+    if (U_SUCCESS(status)) {
```

Comment:
>U_SUCCESS [](start = 8, length = 9)

Question, can the method succeed and set the status flag (e.g to a warning). I recall seeing this in the documentation when i was looking at string tokenizer.